docs/models/advanced/introduction.md

```diff
@@ -0,0 +1,49 @@
+Intro to Advanced Models
+========================
+
+While simple models and blockstates are all well and good, they aren't dynamic. For example, Forge's Universal Bucket can hold all kinds of fluid, which may be mod added. It has to dynamically piece the model together from the base bucket model and the liquid. How is that done? Enter `IModel`.
+
+In order to understand how this works, let's go through the internals of the model system. Throughout this section, you will probably have to refer to this to grasp a clear understanding of what is happening. This is true in reverse as well. You will likely not understand everything happening here, but as you move through this section you should be able to grasp more and more of it until everything is clear.
+
+!!! important
+    If this is your first time reading through do not skip _anything_! It is _imperative_ that you read everything **in order** so as to build a comprehensive understanding! In the same vein, if this is your first time reading, do not follow links if they lead to pages further ahead in the section.
+
+1. A set of `ModelResourceLocation`s are marked as models to be loaded through `ModelLoader`.
+
+    * For items, their models must be manually marked for loading with `ModelLoader.registerItemVariants`. (`ModelLoader.setCustomMRL` does this.)
```

Comment:
setCustomMRL -> setCustomModelResourceLocation

Don't use shorthand when it hasn't been defined